services/agent/src/simcore_service_agent/modules/task_monitor.py

```diff
@@ -0,0 +1,140 @@
+import asyncio
+import logging
+import traceback
+from collections import deque
+from contextlib import suppress
+from dataclasses import dataclass
+from typing import Any, Awaitable, Callable, Coroutine, Final, Optional
+
+from fastapi import FastAPI
+from pydantic import PositiveFloat, PositiveInt
+
+from ..core.settings import ApplicationSettings
+from .volumes_cleanup import backup_and_remove_volumes
+from servicelib.logging_utils import log_context
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_TASK_WAIT_ON_ERROR: Final[PositiveInt] = 10
+
+
+@dataclass
+class _TaskData:
+    callable: Callable
```

Comment:
MINOR: i would rename this variable since ``callable`` is also reserved and in addition it is not very informative...
what about ``start``?

Reply:
Even better, I would do 
```python
@dataclass
class _TaskData:
    coro: Callable 
    args: Any

    async def start():
         return await self.coro(*self.args)

Reply:
Start cannot be used but will use a factory to return a `Coroutine`.
Will also rename callable to `target` as It is used below.